applications/cost-comparing-research-rust-cdt.md

```diff
@@ -0,0 +1,173 @@
+
+## Development Status
+
+### Description of the expected scenarios
+
+```plain
+┌──────────────┐             ┌──────────────┐            ┌─────────────────┐
+│              │  eosio-llvm │              │            │      API        │
+│   EOSIO C++  │  ───────►   │      IR      │   ───────► │    EOSVM IR     │
+│              │             │              │            │ Interface set   │
+└──────────────┘             └──────────────┘            └─────────────────┘
+
+┌──────────────┐  Wasmer's   ┌──────────────┐            ┌─────────────────┐
+│              │ rust-llvm   │              │            │      API'       │
+│ Wasmer Rust  │  ───────►   │     IR'      │   ───────► │   Wasmer IR     │
+│              │             │              │            │ Interface set   │
+└──────────────┘             └──────────────┘            └─────────────────┘
+```
+
+NOTE: There is no official general Rust LLVM. This standardization project is still in incubation status.
+
+Assumptions: For cost-effective building, we want to build from the many works of Wasmer, although Antelope doesn't need to mount Wamser.
+
+#### Cases
+
+1. If `IR ≅ IR'` and `API ≅ API'`
+    - We can directly start from Wasmer's implementation
+        - Implement Antelope-specific data structure and API of `eosio.cdt` (Like table, vector, etc.)
+1. Else if `IR ≅ IR'` and `API != API'`
+    - Implement a compiler, which works for adjusting WASM API interface from Wasmer spec to EOSVM spec
+    - After then, start (1)
+1. Else (`IR != IR'` and `API != API'`)
+    - Mount Wasmer on Antelope would be the cheapest
+    - Mount Wasmer, make compatible with the existing state DB, and work on (1)
+    - [Draft proposal of this case](https://github.com/DELIGHT-LABS/grant-framework/blob/docs/wasmer-integration/applications/wasmer-integration.md)
+
+### Why Wasmer?
+
+Performance of Wasmer is placed in the top tier. See [benchmark 2021](https://00f.net/2021/02/22/webassembly-runtimes-benchmarks/). Also, it has the biggest ecosystem among WASM VM and supports many languages accordingly.
+
+## Development Roadmap
+
+### Milestone Summary
+
+- **Total Estimated Duration:** 1 month
+- **Full-Time Equivalent (FTE):** 2 FTE
+- **Total Costs:** 10,000 USD
+
+### Milestone 1 — Research the difference between EOSVM and Wasmer
+
+- **Estimated duration:** 1 month
+- **FTE:**  2
+- **Costs:** 10,000 USD
+
+| ID  | Deliverable     | Specification                                                                                                            |
+|-----|-----------------|--------------------------------------------------------------------------------------------------------------------------|
+| 0a. | License         | MIT                                                                                                                      |
+| 1   | Research report | Comparison of language-LLVM compilation spec and WASM API spec of EOSVM and Wasmer<br/>A report and a follow-up proposal |
```

Comment:
Can you add some details about what the report will contain? Like the structure, the approaches you take to compare, any code/experiment/sample contracts you run with Wasmer, how we can verify, and conclusions.

Reply:
Hi @linh2931 :)

The feedback is applied on the commit https://github.com/eosnetworkfoundation/grant-framework/pull/83/commits/ad50f04a16ec3160077af869e0780bbf26186eb3